src/wallet/sqlite.cpp

```diff
@@ -432,12 +452,8 @@ void SQLiteBatch::Close()
     };
 
     for (const auto& [stmt_prepared, stmt_description] : statements) {
-        int res = sqlite3_finalize(*stmt_prepared);
-        if (res != SQLITE_OK) {
-            LogPrintf("SQLiteBatch: Batch closed but could not finalize %s statement: %s\n",
-                      stmt_description, sqlite3_errstr(res));
-        }
-        *stmt_prepared = nullptr;
+        // Reset the unique_ptr to destroy the contained SQLiteStatement
+        stmt_prepared->reset();
     }
```

Comment:
The second element in the pair (`stmt_description`) is not used. This snippet can be reduced to:

```cpp
    for (auto stmt : {&m_read_stmt, &m_insert_stmt, &m_overwrite_stmt, &m_delete_stmt, &m_delete_prefix_stmt}) {
        stmt->reset();
    }                   
```

Or simple stupid:
```cpp
    m_read_stmt.reset();
    m_insert_stmt.reset();
    m_overwrite_stmt.reset();
    m_delete_stmt.reset();
    m_delete_prefix_stmt.reset();
```